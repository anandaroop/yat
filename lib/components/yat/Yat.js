'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Yat = function (_Component) {
  _inherits(Yat, _Component);

  function Yat(props) {
    _classCallCheck(this, Yat);

    var _this = _possibleConstructorReturn(this, (Yat.__proto__ || Object.getPrototypeOf(Yat)).call(this, props));

    _this.state = {
      isLocated: false,
      longitude: null,
      latitude: null
    };
    return _this;
  }

  _createClass(Yat, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var _this2 = this;

      navigator.geolocation.getCurrentPosition(function (position) {
        var _position$coords = position.coords,
            longitude = _position$coords.longitude,
            latitude = _position$coords.latitude;

        _this2.setState({
          isLocated: true,
          longitude: longitude,
          latitude: latitude
        });
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _state = this.state,
          isLocated = _state.isLocated,
          longitude = _state.longitude,
          latitude = _state.latitude;

      return _react2.default.createElement(
        'div',
        null,
        isLocated ? longitude + ', ' + latitude : 'locatingâ€¦'
      );
    }
  }]);

  return Yat;
}(_react.Component);

exports.default = Yat;