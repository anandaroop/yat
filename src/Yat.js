import React from 'react'

const Yat = () => {
  return <div>I will be Yat!</div>
}

export default Yat
