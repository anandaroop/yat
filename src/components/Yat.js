import React from 'react'

const Yat = () => {
  return <div>Yat!</div>
}

export default Yat
