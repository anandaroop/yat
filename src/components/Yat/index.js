import React, { Component } from 'react'

const coordinateString = (coordinates, order) => {
  return order === 'lat,lon'
    ? `${coordinates.latitude}, ${coordinates.longitude}`
    : `${coordinates.longitude}, ${coordinates.latitude}`
}
class Yat extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isLocated: false,
      longitude: null,
      latitude: null
    }
  }

  componentDidMount() {
    navigator.geolocation.getCurrentPosition(position => {
      const { longitude, latitude } = position.coords
      this.setState({
        isLocated: true,
        longitude,
        latitude
      })
    })
  }

  render() {
    const { isLocated, longitude, latitude } = this.state
    const { order } = this.props
    return (
      <div>
        {isLocated ? coordinateString({ longitude, latitude }, order) : 'locatingâ€¦'}
      </div>
    )
  }
}

Yat.defaultProps = {
  order: 'lon,lat'
}

export default Yat
