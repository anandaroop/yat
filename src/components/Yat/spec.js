import React from 'react'
import renderer from 'react-test-renderer'
import Yat from '.'

beforeEach(() => {})

it('renders a placeholder while waiting', () => {
  navigator.geolocation = {
    getCurrentPosition: jest.fn()
  }
  const tree = renderer.create(<Yat />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders the returned coordinates', () => {
  const response = {
    coords: { longitude: -90, latitude: 30 }
  }
  navigator.geolocation = {
    getCurrentPosition: jest.fn(cb => cb(response))
  }
  const tree = renderer.create(<Yat />).toJSON()
  expect(tree).toMatchSnapshot()
})


it('can switch the order of the coordinates', () => {
  const response = {
    coords: { longitude: -90, latitude: 30 }
  }
  navigator.geolocation = {
    getCurrentPosition: jest.fn(cb => cb(response))
  }
  const tree = renderer.create(<Yat order="lat,lon" />).toJSON()
  expect(tree).toMatchSnapshot()
})
