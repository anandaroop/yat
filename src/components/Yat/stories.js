import React from 'react'

import { storiesOf } from '@storybook/react'
// import { action } from '@storybook/addon-actions';
// import { linkTo } from '@storybook/addon-links';

import Yat from '.'

storiesOf('Yat', module).add('lon,lat (default)', () => <Yat />)
