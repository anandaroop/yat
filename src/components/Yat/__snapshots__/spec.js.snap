// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can switch the order of the coordinates 1`] = `
<div>
  30, -90
</div>
`;

exports[`renders a placeholder while waiting 1`] = `
<div>
  locatingâ€¦
</div>
`;

exports[`renders the returned coordinates 1`] = `
<div>
  -90, 30
</div>
`;
